Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +59,139 @@ class TransactionHistoryViewController: UIViewController {
-        self.loadTransactions()
+    func applyTxFilter(currentFilter: Int) {
+        refreshControl.showLoader(in: self.tableView)
+        filteredItems.removeAll()
+        
+        switch currentFilter{
+        case 1:
+            self.filterActive = true
+            self.filteredItems = self.transactions.filter{$0.Direction == 0 && $0.Type == GlobalConstants.Strings.REGULAR}
+            self.btnFilter.setTitle(LocalizedStrings.sent.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case 2:
+            self.filterActive = true
+            self.filteredItems = self.transactions.filter{$0.Direction == 1 && $0.Type == GlobalConstants.Strings.REGULAR}
+            self.btnFilter.setTitle(LocalizedStrings.received.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case 3:
+            self.filterActive = true
+            self.filteredItems = self.transactions.filter{$0.Direction == 2 && $0.Type == GlobalConstants.Strings.REGULAR}
+            self.btnFilter.setTitle(LocalizedStrings.yourself.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case 4:
+            self.filterActive = true
+            self.filteredItems = self.transactions.filter{$0.Type == GlobalConstants.Strings.REVOCATION || $0.Type == GlobalConstants.Strings.TICKET_PURCHASE || $0.Type == GlobalConstants.Strings.VOTE}
+            self.btnFilter.setTitle(LocalizedStrings.staking.appending("(\(self.filteredItems.count))"), for: .normal)
+            break
+        case 5:
+            self.filterActive = true
+            self.filteredItems = self.transactions.filter{$0.Type == GlobalConstants.Strings.COINBASE}
+            self.btnFilter.setTitle("COINBASE (\(self.filteredItems.count))", for: .normal)
+            break
+        default:
+            self.filterActive = false
+            self.btnFilter.setTitle(LocalizedStrings.all.appending("(\(self.transactions.count))"), for: .normal)
+            break
+        }
+        self.refreshControl.endRefreshing()
     }
     
     func loadTransactions() {
-        self.mainContens.removeAll()
-        self.Filtercontent.removeAll()
+        self.transactions.removeAll()
+        var error: NSError?
         
-        // use count = 0 to return all transactions
-        AppDelegate.walletLoader.wallet?.transactionHistory(count: 0) { transactions in
-            self.refreshControl.endRefreshing()
-            
-            if transactions == nil || transactions!.count == 0 {
-                self.showNoTransactions()
-                return
-            }
-            
-            self.mainContens = transactions!
-            self.tableView.backgroundView = nil
-            self.tableView.separatorStyle = .singleLine
-            self.reloadTxsForCurrentFilter()
+        let allTransactions = AppDelegate.walletLoader.wallet?.getTransactions(0, txFilter: Int32(0), error: &error)
```

Comment:
Can't just pass `txFilter: 0`?
Ah, that should even use dcrlibwallet constants: `DcrlibwalletTxFilter*`, in this case `DcrlibwalletTxFilterAll`